monkey/monkey_island/cc/agent_event_handlers/update_agent_shutdown_status.py

```diff
@@ -0,0 +1,21 @@
+import logging
+from datetime import datetime
+
+from common.agent_events import AbstractAgentEvent
+from monkey_island.cc.repository import IAgentRepository, StorageError
+
+logger = logging.getLogger(__name__)
+
+
+class update_agent_shutdown_status:
+    def __init__(self, agent_repository: IAgentRepository):
+        self._agent_repository = agent_repository
+
+    def __call__(self, event: AbstractAgentEvent):
+        agent_id = event.source
+        try:
+            agent = self._agent_repository.get_agent_by_id(agent_id)
+            agent.stop_time = datetime.utcfromtimestamp(event.timestamp)
+            self._agent_repository.upsert_agent(agent)
+        except StorageError as err:
```

Comment:
`UnknownStorrageError` for cases if user resets the island somehow while the agents are still running?

Reply:
`UnknownRecordError`

We don't necessarily need to handle either of these, actually. If they're raised, they should trigger flask to respond to the caller with a 500 error.